src/ansys/mapdl/core/mapdl_grpc.py

```diff
@@ -2149,87 +2166,173 @@ def download(
+
         else:
-            target_dir = os.getcwd()
+            raise ValueError(
+                f"The `file` parameter type ({type(files)}) is not supported."
+                "Only strings, tuple of strings or list of strings are allowed."
+            )
 
         for each_file in list_files:
-            try:
-                file_name = os.path.basename(
-                    each_file
-                )  # Getting only the name of the file.
-                #  We try to avoid that when the full path is supplied, it will crash when trying
-                # to do `os.path.join(target_dir"os.getcwd()", file_name "full filename path"`
-                # This will produce the file structure to flat out, but it is find, because recursive
-                # does not work in remote.
-                self._download(
-                    each_file,
-                    out_file_name=os.path.join(target_dir, file_name),
-                    chunk_size=chunk_size,
-                    progress_bar=progress_bar,
+            self._download(
+                each_file,
+                out_file_name=os.path.join(target_dir, each_file),
+                chunk_size=chunk_size,
+                progress_bar=progress_bar,
+            )
+
+        return list_files
+
+    def _validate_files(self, file, extension=None):
+        base_name = os.path.basename(file)
+        self_files = self.list_files()
+
+        if extension is not None:
+            if not isinstance(extension, str):
+                raise TypeError(f"The extension {extension} must be a string.")
+            valid_files = []
+            for filename in self_files:
+                if filename[-len(extension) :] == extension:
+                    valid_files.append(filename)
+            if len(valid_files) > 0:
+                self_files = valid_files
+            else:
+                raise FileNotFoundError(
+                    f"Files with the extension '{extension}' could not be found in the remote session. In a remote session, only the file basename is considered to find in the MAPDL working directory."
                 )
-            except FileNotFoundError:
-                # So far the grpc interface returns size of the file equal
-                # zero, if the file does not exists or its size is zero,
-                # but they are two different things!
-                # In theory, since we are obtaining the files name from
-                # `mapdl.list_files()` they do exist, so
-                # if there is any error, it means their size is zero.
-                pass  # this is not the best.
+            if not "*" in base_name:
+                if extension[0] == ".":
+                    base_name = os.path.splitext(base_name)[0] + extension
+                else:
+                    base_name = os.path.splitext(base_name)[0] + "." + extension
+
+        if base_name in self_files:
+            list_files = [base_name]
```

Comment:
shouldn't you have this inside the `for filename in self_files` loop and have list_files.append(base_name)` so you can take care of multiple files with that extension?

Reply:
Nevermind.